.github/workflows/ci.yml

```diff
@@ -0,0 +1,397 @@
+          # TODO: Replace with artifact deps when that does not require
+          # additional cargo flags.
+          grep -q generate-sample-ledger /tmp/mc-test-packages && \
+            echo '-p mc-util-keyfile' >> /tmp/mc-test-packages || true
+      - name: Run tests
+        uses: ./.github/actions/run-mc-tests
+        with:
+          args: $(cat /tmp/mc-test-packages)
+          junit_xml_filename: junit-mc-tests-${{matrix.runner_index}}.xml
+
+
+  consensus-tests:
+    runs-on: [self-hosted, Linux, large]
+    container: mobilecoin/rust-sgx-base:latest
+
+    strategy:
+      matrix:
+        num_runners: [2]
+        runner_index: [1, 2]
+      # Run each shard to completion.
+      fail-fast: false
+    env:
+      NUM_RUNNERS: ${{ matrix.num_runners }}
+      RUNNER_INDEX: ${{ matrix.runner_index }}
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+      - name: List packages to test
+        run: |
+          cargo test -p 2>&1 | \
+            awk "/mc-consensus/ { print \"-p \" \$1 }" | \
+            sort > /tmp/test-packages
+          split -n l/$RUNNER_INDEX/$NUM_RUNNERS /tmp/test-packages | \
+            tee /tmp/consensus-test-packages
+      - name: Run tests
+        uses: ./.github/actions/run-mc-tests
+        with:
+          args: $(cat /tmp/consensus-test-packages)
+          junit_xml_filename: junit-consensus-tests-${{matrix.runner_index}}.xml
+
+
+  fog-tests:
+    runs-on: [self-hosted, Linux, large]
+    container: mobilecoin/rust-sgx-base:latest
+
+    strategy:
+      matrix:
+        num_runners: [2]
+        runner_index: [1, 2]
+      # Run each shard to completion.
+      fail-fast: false
+    env:
+      NUM_RUNNERS: ${{ matrix.num_runners }}
+      RUNNER_INDEX: ${{ matrix.runner_index }}
+      TEST_DATABASE_URL: postgres://postgres:postgres@postgres
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+      - name: List packages to test
+        run: |
+          cargo test -p 2>&1 | \
+            awk "/mc-fog/ { print \"-p \" \$1 }" | \
+            grep -v mc-fog-ingest | \
+            sort > /tmp/test-packages
+          split -n l/$RUNNER_INDEX/$NUM_RUNNERS /tmp/test-packages | \
+            tee /tmp/fog-test-packages
+          # Hack: mc-fog-distribution needs bins from
+          # mc-util-{keyfile,generate-sample-ledger}.
+          # TODO: Replace with artifact deps when that does not require
+          # additional cargo flags.
+          grep -q fog-distribution /tmp/fog-test-packages && \
+            echo '-p mc-util-keyfile -p mc-util-generate-sample-ledger' >> /tmp/fog-test-packages || true
+      - name: Run tests
+        uses: ./.github/actions/run-mc-tests
+        with:
+          args: $(cat /tmp/fog-test-packages) --build-jobs 4
```

Comment:
why does this one have `--build-jobs` and the others don't?

Reply:
Adding a clarifying comment.